src/libraries/System.Data.Common/ref/System.Data.Common.cs

```diff
@@ -1858,24 +1906,27 @@ public abstract partial class DbCommand : System.ComponentModel.Component, Syste
         System.Data.IDataReader System.Data.IDbCommand.ExecuteReader() { throw null; }
         System.Data.IDataReader System.Data.IDbCommand.ExecuteReader(System.Data.CommandBehavior behavior) { throw null; }
     }
-#nullable disable
     public abstract partial class DbCommandBuilder : System.ComponentModel.Component
     {
         protected DbCommandBuilder() { }
         [System.ComponentModel.DefaultValueAttribute(System.Data.Common.CatalogLocation.Start)]
         public virtual System.Data.Common.CatalogLocation CatalogLocation { get { throw null; } set { } }
         [System.ComponentModel.DefaultValueAttribute(".")]
+        [System.Diagnostics.CodeAnalysis.AllowNull]
         public virtual string CatalogSeparator { get { throw null; } set { } }
         [System.ComponentModel.DefaultValueAttribute(System.Data.ConflictOption.CompareAllSearchableValues)]
         public virtual System.Data.ConflictOption ConflictOption { get { throw null; } set { } }
         [System.ComponentModel.BrowsableAttribute(false)]
         [System.ComponentModel.DesignerSerializationVisibilityAttribute(System.ComponentModel.DesignerSerializationVisibility.Hidden)]
-        public System.Data.Common.DbDataAdapter DataAdapter { get { throw null; } set { } }
+        public System.Data.Common.DbDataAdapter? DataAdapter { get { throw null; } set { } }
         [System.ComponentModel.DefaultValueAttribute("")]
+        [System.Diagnostics.CodeAnalysis.AllowNull]
         public virtual string QuotePrefix { get { throw null; } set { } }
```

Comment:
I see call sites for some of these virtuals checking for null.  Is non-nullable correct for the return value for all of these?

Reply:
[The default implementation](https://github.com/dotnet/runtime/blob/1d6d1d2bbf14493648300d9a562b5dad486b9ca5/src/libraries/System.Data.Common/src/System/Data/Common/DBCommandBuilder.cs#L464) coalesces to string.Empty, and it's supposed to be a string you can always concatenate before/after the string. It's true that it's virtual so anything's possible, though I'm only seeing IsNullOrEmpty in the [Odbc](https://source.dot.net/#System.Data.Odbc/System/Data/Odbc/OdbcCommandBuilder.cs,262) and [OleDb](https://referencesource.microsoft.com/#System.Data/fx/src/data/System/Data/OleDb/OleDbCommandBuilder.cs,332) providers, and [SqlClient just calls base.QuotePrefix](https://github.com/dotnet/SqlClient/blob/master/src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlCommandBuilder.cs#L107).

Where are you seeing null checks, anything persuasive? I can make it nullable if you think that's safer.

Reply:
> Where are you seeing null checks

The IsNullOrEmpty checks you mentioned.  But it's not just in Odbc and OleDb, e.g. DBCommandBuilder allows null in both `private void BuildInformation(DataTable schemaTable)` and `private string QuotedColumn(string column)`.

Reply:
OK, yeah - I see them though they're also IsNullOrEmpty. I think the point is to check for length 0 rather than for null (the coalescing behavior of the property getters is a good indication), but if you think we should annotate these as nullable I can do that. We could also do a more thorough search outside the BCL to see who's overriding DbCommandBuilder and what they're doing, let me know.